sdk/analytics/src/main/java/com/klaviyo/analytics/networking/requests/KlaviyoApiRequest.kt

```diff
@@ -18,88 +18,20 @@ import org.json.JSONObject
  * @property method The [RequestMethod] will determine the type of request being made
  * @property queuedTime The wall clock time that the request was enqueued
  * @property uuid unique identifier of this request
+ *
+ * @constructor
  */
 internal open class KlaviyoApiRequest(
     val urlPath: String,
     val method: RequestMethod,
-    override val queuedTime: Long = Registry.clock.currentTimeMillis(),
-    override val uuid: String = UUID.randomUUID().toString()
+    queuedTime: Long? = null,
+    uuid: String? = null
 ) : ApiRequest {
 
     internal enum class Status {
         Unsent, Inflight, PendingRetry, Complete, Failed
     }
 
-    override val type: String = urlPath
-    override val state: String get() = status.name
-
-    protected var status: Status = Status.Unsent
-        set(value) {
-            if (field == value) return
-            field = value
-
-            if (value == Status.Inflight) {
-                startTime = Registry.clock.currentTimeMillis()
-            } else if (status in arrayOf(Status.Complete, Status.Failed)) {
-                endTime = Registry.clock.currentTimeMillis()
-            }
-        }
-
-    override val httpMethod: String = method.name
-    override var headers: Map<String, String> = emptyMap()
-    override var query: Map<String, String> = emptyMap()
-    open var body: JSONObject? = null
-    override val requestBody: String? get() = formatBody()
-
-    var attempts = 0
-        private set
-
-    override var startTime: Long? = null
-        protected set
-    override var responseCode: Int? = null
-        protected set
-    override var responseBody: String? = null
-        protected set
-    override var endTime: Long? = null
-        protected set
-
-    /**
-     * Creates a representation of this [KlaviyoApiRequest] in JSON
-     *
-     * NOTE: subclass information is lost in the process.
-     * The important data to send the API request is retained.
-     *
-     * @return
-     */
-    fun toJson(): String = JSONObject()
-        .accumulate(PATH_JSON_KEY, urlPath)
-        .accumulate(METHOD_JSON_KEY, method.name)
-        .accumulate(TIME_JSON_KEY, queuedTime)
-        .accumulate(UUID_JSON_KEY, uuid)
-        .accumulate(HEADERS_JSON_KEY, JSONObject(headers))
-        .accumulate(QUERY_JSON_KEY, JSONObject(query))
-        .accumulate(BODY_JSON_KEY, body)
-        .toString()
-
-    override fun toString(): String = toJson()
-
-    open fun formatBody(): String? = body?.let { JSONObject(mapOf(DATA to it)).toString() }
```

Comment:
I had some confusing duplication of logic surrounding the request body. I simplified it a little bit to just a JSON representation and a computed property that translates it to a string.